### PR TITLE
fix: create SA using helm and use SA annotation for EKS

### PR DESCRIPTION
**Pull request by gazal-k**
fix https://github.com/jenkins-x-labs/issues/issues/17

### CODE REVIEW

File: apps/bitnami/external-dns/values.yaml.gotmpl [hunk 1]
```diff
@@ -20,6 +20,9 @@ rbac:
 {{- if eq .Values.jxRequirements.cluster.provider "gke" }}
   serviceAccountAnnotations:
     iam.gke.io/gcp-service-account: {{ .Values.jxRequirements.cluster.clusterName }}-ex@{{ .Values.jxRequirements.cluster.project }}.iam.gserviceaccount.com
+{{- else if eq .Values.jxRequirements.cluster.provider "eks" }}
+  serviceAccountAnnotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::{{ .Values.jxRequirements.cluster.accountId }}:role/{{ .Values.jxRequirements.cluster.clusterName }}-{{ .Values.jxRequirements.cluster.namespace }}-externaldns
```
Comment (gazal-k) [inline apps/bitnami/external-dns/values.yaml.gotmpl]: seems like there's some yaml schema validation in place, can't just add `.Values.jxRequirements.cluster.accountId` and test

File: apps/jenkins-x/jxui/values.yaml.gotmpl [hunk 2]
```diff
@@ -1,7 +1,5 @@
+{{- if eq .Values.jxRequirements.cluster.provider "eks" }}
 serviceaccount:
-{{- if eq "eks" .Values.jxRequirements.cluster.provider }}
-  enabled: false
-{{- else }}
-  enabled: true
+  annotations:
```
Comment (gazal-k) [inline apps/jenkins-x/jxui/values.yaml.gotmpl]: someone will have to add this template variable to the non-OSS jxui chart

File: apps/bitnami/external-dns/values.yaml.gotmpl (hunk 1 above)
Comment (gazal-k) [reply apps/bitnami/external-dns/values.yaml.gotmpl]: verified this using `.Values.jxRequirements.cluster.project` for now.

The only issue we had was it's usage here: https://github.com/jenkins-x-charts/jxboot-helmfile-resources/blob/892d5104ef47d4c7d8cd72e467acb24c869b80ad/jxboot-helmfile-resources/templates/environments.yaml#L44-L48

**Comment by gazal-k** [comment]
this change: https://github.com/jenkins-x/jx/pull/6989 can enable us to provide both options; create IR & SA from cloud tooling or create just IR from cloud tooling and then SA with helm

**Comment by jstrachan** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rawlingsj**
You can assign the PR to them by writing `/assign @rawlingsj` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=jenkins-x-labs%2Fjenkins-x-versions).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/jenkins-x-labs/jenkins-x-versions/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by jstrachan** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/jenkins-x-labs/jenkins-x-versions/pull/41#issuecomment-606855904" title="Approved">rawlingsj</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=jenkins-x-labs%2Fjenkins-x-versions).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/jenkins-x-labs/jenkins-x-versions/blob/master/OWNERS)~~ [rawlingsj]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gazal-k** [comment]
whoa whoa guys, sorry, I don't know if this was quite read. If we can leverage the new config from https://github.com/jenkins-x/jx/pull/6989, we can have both options; create SA using helm or skip SA (all specific to EKS of course)

**Comment by rawlingsj** [comment]
in labs we're working off a branch of jx ATM https://github.com/jenkins-x/jx/tree/multicluster

sorry I didn't realise we needed https://github.com/jenkins-x/jx/pull/6989 I can cherry pick that to our branch in the morning and update the labs repos, would that make sense?

**Comment by gazal-k** [comment]
I understand the `multicluster` long term feature branch thing. We can discuss in detail when you wake up :)

**Pull request merged**